### PR TITLE
[TS migration] Migrate 'useKeyboardState.js' hook to TypeScript

### PR DESCRIPTION
**Pull request by fvlvte**
### Details
Migrated useKeyboardState hook and it's dependency to TypeScript.

### Fixed Issues
$ https://github.com/Expensify/App/issues/24941
$ https://github.com/Expensify/App/issues/24955
PROPOSAL: N/A

### Tests
- [x] Verify that no errors appear in the JS console

### Offline tests
N/A

### QA Steps
- [x] Verify that no errors appear in the JS console
- [x] Click on profile icon from the main screen
- [x] Open profile
- [x] Click on display name
- [x] Focus first name field
- [x] Click on back arrow
- [x] The keyboard is closed by ScreenWrapper that is using useKeyboardState hook.

### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/20582108/960a9883-5630-4621-b33c-dc4197f5beb4

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/20582108/9a7ee70a-169d-4c72-b5b1-ef4ed135429e

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/20582108/d0d46a13-cf77-4b9b-9d93-77cfcd28dbd5

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/20582108/cc91dc99-7f72-4beb-ba06-b97c12891c09

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/20582108/7409f171-3d23-49d2-9688-22632748f059

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/20582108/3293aa7f-c8b1-413c-86b2-0513768d9395

</details>

### CODE REVIEW

File: src/components/withKeyboardState.tsx [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
 
-const keyboardStateProviderPropTypes = {
+type KeyboardStateProviderProps = {
     /* Actual content wrapped by this component */
-    children: PropTypes.node.isRequired,
+    children: ReactNode
 };
 
-function KeyboardStateProvider(props) {
+// TODO: Remove - left for backwards compatibility with existing components.
```
Comment (rezkiy37) [inline src/components/withKeyboardState.tsx:16]: Resolve TODO.

Comment (rezkiy37) [inline src/components/withKeyboardState.tsx:16]: Resolve TODO.

File: src/components/withKeyboardState.tsx [hunk 2]
```diff
@@ -1,20 +1,25 @@
...
+const keyboardStatePropTypes = {
+    /** Whether the keyboard is open */
+    isKeyboardShown: PropTypes.bool.isRequired,
+};
+const KeyboardStateContext = createContext<KeyboardStateContextValue | null>(null);
+function KeyboardStateProvider(props: KeyboardStateProviderProps) {
```
Comment (rezkiy37) [inline src/components/withKeyboardState.tsx:22]: Add empty lines between different kind of blocks.
```suggestion
const keyboardStatePropTypes = {
    /** Whether the keyboard is open */
    isKeyboardShown: PropTypes.bool.isRequired,
};

const KeyboardStateContext = createContext<KeyboardStateContextValue | null>(null);

function KeyboardStateProvider(props: KeyboardStateProviderProps) {
```

File: src/components/withKeyboardState.tsx [hunk 3]
```diff
@@ -34,14 +39,10 @@ function KeyboardStateProvider(props) {
...
 
-KeyboardStateProvider.propTypes = keyboardStateProviderPropTypes;
 
-/**
- * @param {React.Component} WrappedComponent
- * @returns {React.Component}
- */
-export default function withKeyboardState(WrappedComponent) {
-    const WithKeyboardState = forwardRef((props, ref) => (
+// eslint-disable-next-line @typescript-eslint/naming-convention
```
Comment (rezkiy37) [inline src/components/withKeyboardState.tsx:43]: Why have you added this, because it wasn't before?

**Review by rezkiy37 — CHANGES_REQUESTED** [review]
Left comments.

File: src/components/withKeyboardState.tsx (hunk 1 above)
Comment (rezkiy37) [reply src/components/withKeyboardState.tsx:16]: So, this TODO is waiting for another PR?

Comment (rezkiy37) [reply src/components/withKeyboardState.tsx:16]: Okay. Do we have an issue where we can put it is like a requirement to don't forget?

File: src/components/withKeyboardState.tsx [hunk 4]
```diff
@@ -52,11 +55,9 @@ export default function withKeyboardState(WrappedComponent) {
                     ref={ref}
                 />
             )}
-        </KeyboardStateContext.Consumer>
-    ));
-
-    WithKeyboardState.displayName = `withKeyboardState(${getComponentDisplayName(WrappedComponent)})`;
+        </KeyboardStateContext.Consumer>));
+    (WithKeyboardState as unknown as {displayName: string}).displayName = `withKeyboardState(${getComponentDisplayName(WrappedComponent as ComponentType)})`;
```
Comment (VickyStash) [inline src/components/withKeyboardState.tsx:59]: @fvlvte You can add a type to `WithKeyboardState` above so you can get rid of type assertions here. 
Also, have you tested this HOC type? maybe create a simple TS component, wrap it with HOC, and see if typing is ok.

File: src/components/withKeyboardState.tsx [hunk 5]
```diff
@@ -34,14 +41,9 @@ function KeyboardStateProvider(props) {
...
-KeyboardStateProvider.propTypes = keyboardStateProviderPropTypes;
-
-/**
- * @param {React.Component} WrappedComponent
- * @returns {React.Component}
- */
-export default function withKeyboardState(WrappedComponent) {
-    const WithKeyboardState = forwardRef((props, ref) => (
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export default function withKeyboardState(WrappedComponent: ComponentType<{ref: ForwardedRef<unknown>}>) {
```
Comment (blazejkustra) [inline src/components/withKeyboardState.tsx:45]: Add return type for this function, similar how Bartek did [here](https://github.com/Expensify/App/pull/29426/files)

File: src/components/withKeyboardState.tsx [hunk 6]
```diff
@@ -1,20 +1,27 @@
...
+type KeyboardStateContextValue = {
+    /** Whether the keyboard is open */
+    isKeyboardShown: boolean;
 };
 
-const keyboardStateProviderPropTypes = {
+type KeyboardStateProviderProps = {
     /* Actual content wrapped by this component */
-    children: PropTypes.node.isRequired,
+    children: ReactNode;
```
Comment (VickyStash) [inline src/components/withKeyboardState.tsx:13]: You can use `ChildrenProps` instead

File: src/components/withKeyboardState.tsx [hunk 7]
```diff
@@ -40,14 +47,9 @@ function KeyboardStateProvider(props) {
...
 
-KeyboardStateProvider.propTypes = keyboardStateProviderPropTypes;
-
-/**
- * @param {React.Component} WrappedComponent
- * @returns {React.Component}
- */
-export default function withKeyboardState(WrappedComponent) {
-    const WithKeyboardState = forwardRef((props, ref) => (
+// eslint-disable-next-line @typescript-eslint/naming-convention
```
Comment (VickyStash) [inline src/components/withKeyboardState.tsx:50]: No need in this eslint disable

File: src/components/withKeyboardState.tsx [hunk 8]
```diff
@@ -40,14 +47,9 @@ function KeyboardStateProvider(props) {
...
+export default function withKeyboardState(WrappedComponent: ComponentType<{ref: ForwardedRef<unknown>}>) {
```
Comment (VickyStash) [inline src/components/withKeyboardState.tsx:51]: For HOC typing, please follow the logic other HOCs use. Take a look at `withLocalize` HOC as a reference.
1. Add HOC return type
2. Use generics
3. Update forwardRef logic

File: src/components/withKeyboardState.tsx [hunk 9]
```diff
@@ -60,9 +62,8 @@ export default function withKeyboardState(WrappedComponent) {
             )}
         </KeyboardStateContext.Consumer>
     ));
-
-    WithKeyboardState.displayName = `withKeyboardState(${getComponentDisplayName(WrappedComponent)})`;
+    (WithKeyboardState as unknown as {displayName: string}).displayName = `withKeyboardState(${getComponentDisplayName(WrappedComponent as ComponentType)})`;
```
Comment (VickyStash) [inline src/components/withKeyboardState.tsx:65]: Get rid of assertions. Use `withLocalize` HOC as a reference.

File: src/components/withKeyboardState.tsx [hunk 10]
```diff
@@ -60,9 +62,8 @@ export default function withKeyboardState(WrappedComponent) {
...
     return WithKeyboardState;
 }
 
-export {KeyboardStateProvider, keyboardStatePropTypes, KeyboardStateContext};
+export {KeyboardStateProvider, keyboardStatePropTypes, type KeyboardStateContextValue, KeyboardStateContext};
```
Comment (VickyStash) [inline src/components/withKeyboardState.tsx:69]:
```suggestion
export {KeyboardStateProvider, keyboardStatePropTypes, KeyboardStateContext};
export type {KeyboardStateContextValue};
```

**Review by VickyStash — CHANGES_REQUESTED** [review]

File: src/components/withKeyboardState.tsx [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+                        // eslint-disable-next-line react/jsx-props-no-spreading
+                        {...keyboardStateProps}
+                        // eslint-disable-next-line react/jsx-props-no-spreading
+                        {...props}
+                        ref={ref}
+                    />
+                )}
+            </KeyboardStateContext.Consumer>
+        );
+    }
```
Comment (VickyStash) [inline src/components/withKeyboardState.tsx:60]: I think `TProps` should extend the context value type. Also please, test that typing works as expected
```suggestion
export default function withKeyboardState<TProps extends KeyboardStateContextValue, TRef>(WrappedComponent: ComponentType<TProps & RefAttributes<TRef>>): (props: Omit<TProps, keyof KeyboardStateContextValue> & React.RefAttributes<TRef>) => ReactElement | null {
    function WithKeyboardState(props: Omit<TProps, keyof KeyboardStateContextValue>, ref: ForwardedRef<TRef>) {
        return (
            <KeyboardStateContext.Consumer>
                {(keyboardStateProps) => (
                    <WrappedComponent
                        // eslint-disable-next-line react/jsx-props-no-spreading
                        {...keyboardStateProps}
                        // eslint-disable-next-line react/jsx-props-no-spreading
                        {...(props as TProps)}
                        ref={ref}
                    />
                )}
            </KeyboardStateContext.Consumer>
        );
    }
```

**Review by VickyStash — CHANGES_REQUESTED** [review]

**Review by fabioh8010 — APPROVED** [review]

**Review by blazejkustra — APPROVED** [review]

**Comment by situchan** [comment]
Why is this No QA? Keyboard handling is important part of the app. Please add Test/QA step at least for one page affected by this change

**Comment by fabioh8010** [comment]
@fvlvte Please add test steps for this one!

**Comment by fabioh8010** [comment]
@situchan Bump

**Comment by situchan** [comment]
will review today. please merge main

**Comment by fabioh8010** [comment]
@situchan It's merged now

**Comment by fabioh8010** [comment]
@situchan friendly bump

File: src/components/withKeyboardState.tsx (hunk 1 above)
Comment (situchan) [reply src/components/withKeyboardState.tsx:12]: Not in this PR about how about adding lint rule to disallow propTypes definition in TS?
Then temporarily disable lint on this line and after relevant components are migrated, we can remove.

**Comment by situchan** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



https://github.com/Expensify/App/assets/108292595/c54b47d0-e75b-4b95-8f39-01837bfc215d


</details>

<details>
<summary>Android: mWeb Chrome</summary>


https://github.com/Expensify/App/assets/108292595/4eae408e-936c-4664-b42e-e5f587046f56



</details>

<details>
<summary>iOS: Native</summary>


https://github.com/Expensify/App/assets/108292595/4bdea51c-f957-48d0-ad02-802c3120af85



</details>

<details>
<summary>iOS: mWeb Safari</summary>


https://github.com/Expensify/App/assets/108292595/8243e650-d809-4e42-b01f-564236d5811f



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


https://github.com/Expensify/App/assets/108292595/7f66af9b-3226-4294-abd1-6393ee9175dc



</details>

<details>
<summary>MacOS: Desktop</summary>



https://github.com/Expensify/App/assets/108292595/c40a136e-32f8-4f73-9eb0-95a945ca00ef


</details>

**Review by situchan — APPROVED** [review]

**Review by situchan — APPROVED** [review]
Re-approving to get engineer review

**Comment by VickyStash** [comment]
@lakchote kind bump 🙂

**Comment by mountiny** [comment]
@situchan sorry for the bother, could you give this one quick test after @fvlvte syncs with main, its been two weeks since tested so I want to make sure nothing popped up since

**Comment by situchan** [comment]
Tests well!

File: src/components/withKeyboardState.tsx [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+import {Keyboard} from 'react-native';
+import getComponentDisplayName from '@libs/getComponentDisplayName';
+import ChildrenProps from '@src/types/utils/ChildrenProps';
+
+type KeyboardStateContextValue = {
+    /** Whether the keyboard is open */
+    isKeyboardShown: boolean;
+};
+
+// TODO: Remove - left for backwards compatibility with existing components.
```
Comment (mountiny) [inline src/components/withKeyboardState.tsx:12]: Is this still required, if so, can you link an issue/ PR which is tied to this?

**Review by mountiny — DISMISSED** [review]
@fvlvte One comment, I think the TODO is too generic and also I see that its been in place for couple months so is it indeed still pending?

**Comment by mountiny** [comment]
@fvlvte Can you link https://github.com/Expensify/App/issues/25151 in the comment there please?

**Review by mountiny — APPROVED** [review]
Appreciate it!

**Pull request merged**